### DISCUSSION TITLE
Issue when testing component that has temporaryUrl except specifc extensions

### DISCUSSION THREAD

**Discussion by jimmypuckett** [livewire/livewire#1591] [3. Feature ideas] [answered]
### Description

Geting an error on temporaryUrl for sideloaded pdfs

```
Tests\Feature\Forms\FileUploadTest::it_allows_specific_file_types with data set "pdf" ('pdf', 'create', true)
ErrorException: This driver does not support creating temporary URLs. (View: /some/path/resources/views/components/forms/upload.blade.php) (View: /some/path/resources/views/components/forms/upload.blade.php)
```

### Steps to reproduce

We have a blade that has a preview of the sideloaded file...

```php
@if($upload and ! $errors->has('upload'))
                <embed
                    src="{{ $upload->temporaryUrl() }}"
                    type="{{ $upload->getMimeType() }}"
                    width="100%"
                    height="500px"
                    class="my-4"
                />
@endif
```

Since the `env` is testing, then is goes into the check to see if supported type, which it is not, so it has passes the request to storage...

https://github.com/livewire/livewire/blob/8247b1fd7221624ce2d58a08135a68666baea9d9/src/TemporaryUploadedFile.php#L79

I know that I can mock all of this out, but that just seems fragile as I have to know the exact methods inside of livewire that is calling to storage to mock out.

If `pdf` was added to the built in supported files, this would fix my issue, but that is not really a good fix for other extensions.

What is the preferred solution?

**Context**
- Livewire version: 1.3.3
- Laravel version: 7.28.1
- Browser: CLI

**Comment by jimmypuckett** [comment]
What if the `supportedPreviewType` was set in the config?  Then a user could override?

**Comment by calebporzio** [comment, marked as answer]
This is now configurable in Livewire's config file: `livewire.temporary_file_upload.preview_mimes` -> you can add any you need there for support!

**Reply by bjornbjorn** [reply]
How does this work though? If you add 'pdf' to that array in the config, the app won't bail with an exception anymore, but you'll still get a broken image as LiveWire doesn't generate previews of PFDs -- is it possible to hook onto this part and serve the thumbnail of the PDF?

**Reply by ShekhSaifuddin007** [reply]
@calebporzio 

Still, this gives us this error, if we add `.pdf`/`.txt` or something else which not an image.

`This driver does not support creating temporary URLs`

**Reply by bjornbjorn** [reply]
@ShekhSaifuddin007  I'm able to remove this error message if I add ".pdf" to the config.

It occurred to me that the temporaryUrl's aren't previews per se, it's basically the full file being sent back, so if you uploaded a .PDF you'll get access to the full file, same with the images.
